BossMod/BossModule/AOEShapes.cs

```diff
@@ -234,4 +234,61 @@ private IEnumerable<WPos> ContourPoints(WPos origin, Angle rotation, float offse
             yield return origin + dx1 + dy2;
         }
     }
+
+    public class AOEShapeTriangle : AOEShape
+    {
+        public float SideLength;
+        public Angle DirectionOffset;
+
+        public AOEShapeTriangle(float sideLength, Angle directionOffset = new())
+        {
+            SideLength = sideLength;
+            DirectionOffset = directionOffset;
+        }
+
+        public override bool Check(WPos position, WPos origin, Angle rotation)
+        {
+            var vertices = CalculateVertices(origin, rotation + DirectionOffset);
+            return position.InTri(vertices.p1, vertices.p2, vertices.p3);
+        }
+
+        public override void Draw(MiniArena arena, WPos origin, Angle rotation, uint color = ArenaColor.AOE)
+        {
+            var vertices = CalculateVertices(origin, rotation + DirectionOffset);
+            arena.AddTriangleFilled(vertices.p1, vertices.p2, vertices.p3, color);
+        }
+
+        public override void Outline(MiniArena arena, WPos origin, Angle rotation, uint color = ArenaColor.Danger)
+        {
+            var vertices = CalculateVertices(origin, rotation + DirectionOffset);
+            arena.AddTriangle(vertices.p1, vertices.p2, vertices.p3, color);
+        }
+
+        public override IEnumerable<IEnumerable<WPos>> Contour(WPos origin, Angle rotation, float offset = 0, float maxError = 1)
+        {
+            var vertices = CalculateVertices(origin, rotation + DirectionOffset, offset);
+            return new List<IEnumerable<WPos>> { new[] { vertices.p1, vertices.p2, vertices.p3 } };
+        }
+
+        public override Func<WPos, float> Distance(WPos origin, Angle rotation)
+        {
+            // Implementing an exact distance calculation for a triangle shape might be complex and is beyond the scope of this basic implementation.
+            return p => (p - origin).Length(); // Simplified placeholder
```

Comment:
This is just an intersection (== max) of 3 half-planes. Very similar to ShapeDistance.Rect, with different optimization opportunities.